### ISSUE TITLE
Singular Value Decomposition (cv::SVD::compute(A, w, u, vt)) may generate incorrect calculated left singular vectors(u).

### ISSUE THREAD

**Issue by xvvzhang** [opencv/opencv#20975]
##### System information (version)

- OpenCV => 4.5.4-dev
- Operating System / Platform =>  Ubuntu
- Compiler => g++

##### Detailed description
Singular Value Decomposition (cv::SVD::compute(A, w, u, vt)) may generate incorrect calculated left singular vectors(u).

For example, when the input matrix A =
[0, 0, 0;
 1, 0, 0;
 0, 0, 0]

The calculated left singular vectors, u, should be
[0, -1, 0;
-1, 0, 0;
 0, 0, 1]

However, the actual output is 
[0, -0.70710677, -0.70710677;
 1, 0, 0;
 0, 0.70710677, -0.70710677]

Another example: 
Input matrix A =
[0, 0, 0;
 1, 0, 1;
 0, 0, 0]

Expected output u =
[0, -1, 0;
 -1, 0, 1;
 0, 0, 1]

The actual output u =
[0, -0.70710677, -0.70710677;
 1,               0,                0        ;
 0, 0.70710677, -0.70710677]

##### Steps to reproduce
```
#include <iostream>
#include <opencv2/opencv.hpp>

using namespace std;
using namespace cv;

int main() {

    cv::Mat A = cv::Mat(3, 3, CV_32F)
    = (cv::Mat_<float>(3, 3, CV_32F)
       <<
       0, 0, 0,
       1, 0, 0,
       0, 0, 0);
    
    Mat u, w, vt;
    SVD::compute(A, w, u, vt);

    cout << "u:" << endl;
    cout << u << endl << endl;

    return 0;
}
```

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues,
       forum.opencv.org, Stack Overflow, etc and have not found solution
   
 - [x] I updated to latest OpenCV version and the issue is still there
   
 - [x] There is reproducer code and related data files: videos, images, onnx, etc

**Comment by xvvzhang** [comment]
I found this problem when I tried to sample inside hyperellipsoids uniformly. I followed the method described in [Informed RRT*: Optimal Incremental Path Planning Focused through an Admissible Ellipsoidal Heuristic](https://www.researchgate.net/publication/261512737_Informed_RRT_Optimal_Incremental_Path_Planning_Focused_through_an_Admissible_Ellipsoidal_Heuristic). However, with cv::SVD::compute(), I got minor axes are .5^.5 of the correct value. Sometimes minor axes collapse into zero.
Then I tied Numpy, Eigen, Octave. They work well to produce the correct result.

**Comment by alalek** [comment]
> The calculated left singular vectors, u, **should be**

The SVD is not unique in general.

Which property or equation of SVD is broken?

**Comment by xvvzhang** [comment]
Yes. The SVD is not unique. But, for singular values equal to zero, their singular vectors may contain some strange things like 0.70710677(.5^.5). These values don't matter for pseudo-inverse or matrix recovery because the singular values are zero. However, they are not negligible for independent u and v operations. I suppose the problem is from JacobiImpl_ of the [source code](https://github.com/opencv/opencv/blob/master/modules/core/src/lapack.cpp).

**Comment by AleksandrPanov** [comment]
> > > The calculated left singular vectors, u, **should be**
> > 
> > 
> > The SVD is not unique in general.
> > Which property or equation of SVD is broken?
> 
> Sorry for my imprecise description. The SVD is not unique. But, for singular values equal to zero, their singular vectors may contain some strange things like 0.70710677(.5^.5). These values don't matter for pseudo-inverse or matrix recovery because the singular values are zero. However, they are not negligible for independent u and v operations. I suppose the problem is from [JacobiImpl_](https://github.com/opencv/opencv/blob/master/modules/core/src/lapack.cpp). The Eigen::JacobiSVD also works well for me.

Indeed, there are no problems in recovering matrix A.
`cv::Mat recoveredA = u * W * vt;`
[recoverA.cpp.txt](https://github.com/opencv/opencv/files/7469446/recoverA.cpp.txt)

Why is the value 0.70710677(.5^.5) strange? What is the problem in your opinion?

**Comment by xvvzhang** [comment]
> > > > The calculated left singular vectors, u, **should be**
> > > 
> > > 
> > > The SVD is not unique in general.
> > > Which property or equation of SVD is broken?
> > 
> > 
> > Sorry for my imprecise description. The SVD is not unique. But, for singular values equal to zero, their singular vectors may contain some strange things like 0.70710677(.5^.5). These values don't matter for pseudo-inverse or matrix recovery because the singular values are zero. However, they are not negligible for independent u and v operations. I suppose the problem is from [JacobiImpl_](https://github.com/opencv/opencv/blob/master/modules/core/src/lapack.cpp). The Eigen::JacobiSVD also works well for me.
> 
> Indeed, there are no problems in recovering matrix A. `cv::Mat recoveredA = u * W * vt;` [recoverA.cpp.txt](https://github.com/opencv/opencv/files/7469446/recoverA.cpp.txt)
> 
> Why is the value 0.70710677(.5^.5) strange? What is the problem in your opinion?

The result of SVD::compute cannot reproduce a unique matrix mentioned in the [paper](https://www.researchgate.net/publication/261512737_Informed_RRT_Optimal_Incremental_Path_Planning_Focused_through_an_Admissible_Ellipsoidal_Heuristic) by u and v.

The matrix is C = U * diag { 1, . . . , 1, det (U) det (V) } * V.t().

I am willing to share the code as well as its visualization, and debug the SVD code together.
Should I post my code here?

**Comment by xvvzhang** [comment]
> > > > The calculated left singular vectors, u, **should be**
> > > 
> > > 
> > > The SVD is not unique in general.
> > > Which property or equation of SVD is broken?
> > 
> > 
> > Sorry for my imprecise description. The SVD is not unique. But, for singular values equal to zero, their singular vectors may contain some strange things like 0.70710677(.5^.5). These values don't matter for pseudo-inverse or matrix recovery because the singular values are zero. However, they are not negligible for independent u and v operations. I suppose the problem is from [JacobiImpl_](https://github.com/opencv/opencv/blob/master/modules/core/src/lapack.cpp). The Eigen::JacobiSVD also works well for me.
> 
> Indeed, there are no problems in recovering matrix A. `cv::Mat recoveredA = u * W * vt;` [recoverA.cpp.txt](https://github.com/opencv/opencv/files/7469446/recoverA.cpp.txt)
> 
> Why is the value 0.70710677(.5^.5) strange? What is the problem in your opinion?

My guess is probably right.
In [line 547](https://github.com/opencv/opencv/blob/master/modules/core/src/lapack.cpp), it comments:
            // if we got a zero singular value, then in order to get the corresponding left singular vector
            // we generate a random vector, project it to the previously computed left singular vectors,
            // subtract the projection and normalize the difference.
